### PR TITLE
Remove tar dependency and used platform specific dependencies for different formats

### PR DESCRIPTION
**Pull request by VirtualPirate**

### CODE REVIEW

**Review by nathanbabcock — DISMISSED** [review]
@VirtualPirate Works like a charm locally for me on Windows, and CI passed on all other platforms. Love how concise it is, too.

Thanks for your work on this and the previous PR. Please feel free to merge :)

File: Cargo.toml [hunk 1]
```diff
@@ -17,7 +17,13 @@ download_ffmpeg = ["reqwest"]
 [lib]
 crate-type = ["lib"]
 
+[target.'cfg(target_os = "linux")'.dependencies]
+tar = "0.4.42"
```
Comment (nathanbabcock) [inline Cargo.toml:21]: Oh yeah, one other small thing. What do you think about making these 3 deps optional, and grouping them all underneath the `download_ffmpeg` feature?

My idea is that if someone has opted out of the download functionality, maybe they have their own plans about how to acquire and extract the binary and they don't need these dependencies. So the `download_ffmpeg` feature could look like: `download_ffmpeg = ["reqwest", "zip", "tar", "xz2"]`. Do you agree?

Comment (VirtualPirate) [reply Cargo.toml:21]: Affirmative, makes more sense actually. Let me make this change

**Review by nathanbabcock — APPROVED** [review]

**Comment by VirtualPirate** [comment]
@nathanbabcock you can merge it, if everything looks fine. I don't have write access to the repo

**Pull request merged**